core/src/test/java/org/bitcoinj/core/FloLegacyAddressTest.java

```diff
@@ -0,0 +1,271 @@
+                .readObject();
+        assertEquals(mainAddress, mainAddressCopy);
+    }
+
+    @Test
+    public void stringification() throws Exception {
+        // Test a testnet address.
+        LegacyAddress a = LegacyAddress.fromPubKeyHash(FLOTESTNET, HEX.decode("aaa17faff597a3a5d20ba5564b2e7104cf2d62ed"));
+        assertEquals("oYRaaENhPh4BM3Ync7xACL7z2LxMxA5z2R", a.toString());
+        assertEquals(ScriptType.P2PKH, a.getOutputScriptType());
+
+        LegacyAddress b = LegacyAddress.fromPubKeyHash(FLOMAINNET, HEX.decode("222fe16d40ff72c2337d07748277d42d8cd0ce2f"));
+        assertEquals("F8wso89xU3FhBVkBsZc6RNdA1x8YFCE6xi", b.toString());
+        assertEquals(ScriptType.P2PKH, a.getOutputScriptType());
+    }
+    
+    @Test
+    public void decoding() throws Exception {
+        LegacyAddress a = LegacyAddress.fromBase58(FLOTESTNET, "oYRaaENhPh4BM3Ync7xACL7z2LxMxA5z2R");
+        assertEquals("aaa17faff597a3a5d20ba5564b2e7104cf2d62ed", Utils.HEX.encode(a.getHash()));
+
+        LegacyAddress b = LegacyAddress.fromBase58(FLOMAINNET, "F8wso89xU3FhBVkBsZc6RNdA1x8YFCE6xi");
+        assertEquals("222fe16d40ff72c2337d07748277d42d8cd0ce2f", Utils.HEX.encode(b.getHash()));
+    }
+    
+    @Test
+    public void errorPaths() {
+
+        // Check the case of a mismatched network.
+        try {
+            LegacyAddress.fromBase58(FLOTESTNET, "F8wso89xU3FhBVkBsZc6RNdA1x8YFCE6xi");
+            fail();
+        } catch (AddressFormatException.WrongNetwork e) {
+            // Success.
+        } catch (AddressFormatException e) {
+            fail();
+        }
+
+        // Check the empty case.
+        try {
+            LegacyAddress.fromBase58(FLOTESTNET, "");
+            fail();
+        } catch (AddressFormatException.WrongNetwork e) {
+            fail();
+        } catch (AddressFormatException e) {
+            // Success.
+        }
+
+        // Check the case of a mismatched network.
+        try {
+            LegacyAddress.fromBase58(FLOTESTNET, "17kzeh4N8g49GFvdDzSf8PjaPfyoD1MndL");
+            fail();
+        } catch (AddressFormatException.WrongNetwork e) {
+            // Success.
+        } catch (AddressFormatException e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void getNetwork() throws Exception {
+        NetworkParameters params = LegacyAddress.getParametersFromAddress("F8wso89xU3FhBVkBsZc6RNdA1x8YFCE6xi");
+        assertEquals(FLOMAINNET.getId(), params.getId());
+        params = LegacyAddress.getParametersFromAddress("oYRaaENhPh4BM3Ync7xACL7z2LxMxA5z2R");
+        assertEquals(FLOTESTNET.getId(), params.getId());
+    }
+
+    @Test
+    public void p2shAddress() throws Exception {
+
+        /**
+         * Note that there is something very wrong because both mainnet and testnet values are the same!
+         * See method p2shAddressCreationFromKeys() and fix this once those values are fixed.
```

Comment:
please log a jira ticket for this bug (if there isn't one already)

Reply:
Good call!  https://mediciventures.atlassian.net/browse/DS-69

Reply:
Done: https://mediciventures.atlassian.net/browse/DS-69